### PR TITLE
Username Check

### PR DESCRIPTION
**Pull request by noahnovickf**

### CODE REVIEW

File: src/containers/profile/profile.component.jsx [hunk 1]
```diff
@@ -9,19 +11,25 @@ const Profile = (props) => {
...
-      alert("Please enter a username and upload an avatar");
-    } else {
-      const userWithUsername = {
-        id: auth.currentUser.uid,
-        username: username,
-        user: userFromState,
-      };
-      addUsernameToStateThunk(userWithUsername);
-      uploadProfileAvatarThunk({ image: imageAsFile, username });
+    usernameCheck(username).then((res) => {
```
Comment (lolcode8) [inline src/containers/profile/profile.component.jsx:14]: username check not a very clear function name. Best add comment above the function to explain what the function does. I think there are some examples of this in the codebase

File: src/containers/profile/profile.component.jsx [hunk 2]
```diff
@@ -9,19 +11,25 @@ const Profile = (props) => {
...
+            id: auth.currentUser.uid,
+            username: username,
+            user: userFromState,
+          };
+          addUsernameToStateThunk(userWithUsername);
+          uploadProfileAvatarThunk({ image: imageAsFile, username });
 
-      //TODO: Need to show user that this is successful or has failed
-    }
+          //TODO: Need to show user that this is successful or has failed
```
Comment (lolcode8) [inline src/containers/profile/profile.component.jsx:27]: Is this a nice to have for later?

File: src/database/usernameCheck.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  const snapshot = await db.collection("users").get();
+  const usernameArr = [];
+  snapshot.forEach((user) => {
+    usernameArr.push(user.data().username);
+    // if (user.data().username === u) {
+    //   return false;
+    // } else {
+    //   return true;
+    // }
+  });
```
Comment (lolcode8) [inline src/database/usernameCheck.js:13]: What dis?

File: src/database/usernameCheck.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { db } from "database";
+
+export const usernameCheck = async (u) => {
```
Comment (lolcode8) [inline src/database/usernameCheck.js:3]: What is u? user or username?

File: src/containers/profile/profile.component.jsx (hunk 2 above)
Comment (noahnovickf) [reply src/containers/profile/profile.component.jsx:27]: I feel like they get redirected ... thats enough?

File: src/database/usernameCheck.js (hunk 3 above)
Comment (noahnovickf) [reply src/database/usernameCheck.js:13]: dw about it

File: src/database/usernameCheck.js (hunk 4 above)
Comment (noahnovickf) [reply src/database/usernameCheck.js:3]: username, changed to username

Comment (noahnovickf) [reply src/database/usernameCheck.js:3]: potentialUsername***

**Review by lolcode8 — APPROVED** [review]

**Pull request merged**